drivers/storage/ebs/storage/ebs_storage.go

```diff
@@ -77,43 +81,70 @@ func (d *driver) Init(context types.Context, config gofig.Config) error {
 }
 
 var (
-	sessions    = map[string]interface{}{}
-	sessionsRWL = &sync.RWMutex{}
+	sessions  = map[string]*awsec2.EC2{}
+	sessionsL = &sync.Mutex{}
 )
 
-func (d *driver) Login(ctx types.Context) (interface{}, error) {
-
-	fields := map[string]interface{}{
-		"accessKey": d.accessKey(),
-		"tag":       d.tag(),
+func writeHkey(h hash.Hash, ps *string) {
+	if ps == nil {
+		return
 	}
+	h.Write([]byte(*ps))
+}
 
-	log.WithFields(fields).Debug("beginning ebs login attempt")
-
-	if d.secretKey() == "" {
-		fields["secretKey"] = ""
-	} else {
-		fields["secretKey"] = "******"
-	}
+func (d *driver) Login(ctx types.Context) (interface{}, error) {
+	sessionsL.Lock()
+	defer sessionsL.Unlock()
 
 	var (
-		region   *string
 		endpoint *string
-		sess     = session.New()
+		ckey     string
+		hkey     = md5.New()
+		akey     = d.accessKey
+		region   = d.mustRegion(ctx)
 	)
 
-	if region = d.mustRegion(ctx); region != nil {
-		fields["region"] = *region
+	if region != nil {
```

Comment:
Is this possible after the use of `d.mustRegion()` above? That function makes sure region is set in the IID fields, and that is is not an empty string (`""`), but can it still be `nil`?